src/AppInstallerCLITests/WorkFlow.cpp

```diff
@@ -815,6 +815,29 @@ TEST_CASE("InstallFlow_InstallationNotes", "[InstallFlow][workflow]")
     REQUIRE(installOutput.str().find("testInstallationNotes") != std::string::npos);
 }
 
+TEST_CASE("InstallFlow_UnsupportedArguments", "[InstallFlow][workflow]")
+{
+    TestCommon::TempFile installResultPath("TestExeInstalled.txt");
+
+    std::ostringstream installOutput;
+    TestContext context{ installOutput, std::cin };
+    auto previousThreadGlobals = context.SetForCurrentThread();
+    OverrideForShellExecute(context);
+    context.Args.AddArg(Execution::Args::Type::Manifest, TestDataFile("InstallFlowTest_UnsupportedArguments.yaml").GetPath().u8string());
+    context.Args.AddArg(Execution::Args::Type::Log);
+    context.Args.AddArg(Execution::Args::Type::InstallLocation);
+
+    InstallCommand install({});
+    install.Execute(context);
+    INFO(installOutput.str());
+
+    // Verify Unsupported arguments error message is shown
+    REQUIRE(context.GetTerminationHR() == S_OK);
+    REQUIRE(std::filesystem::exists(installResultPath.GetPath()));
+    REQUIRE(installOutput.str().find(Resource::LocString(Resource::String::UnsupportedArgument).get() + " --log") != std::string::npos);

```

Comment:
Nit: I think a test to check that this is not printed when not needed would be good. E.g. if it is supported, or if it is unsupported but not used. But it's probably not needed.

Reply:
Added another unit test to ensure error messages are not shown without the provided arguments